### PR TITLE
Jinchao add weekly summary indicator on leaderboard for admin users

### PR DESCRIPTION
**Pull request by suigunfjc**
# Description
image

This PR adds a green √ next to the green/red dot/star on leaderboard for users who have submitted their weekly summaries of this week.

## Related PRS (if any):
Invisibility update PR: #859 
This frontend PR is related to the [#383](https://github.com/OneCommunityGlobal/HGNRest/pull/383) backend PR, please test these two PRs together.

## Main changes explained:
- Moved `onClick={() => dashboardToggle(item)}` from outside `<td>` to inside `<div>` that wrapped the dot/star.
- Added a flex div to hold the weekly summary indicator and the state dot/star, which will keep the dot/star on left for admin user and in center for non-admin user.
- Added the green check mark.

## How to test:
1. check into current branch on both HGNRest and HGNApp
2. do `npm run build` and `npm start` on HGNRest
3. log as Owner/Admin/Core Team user and check if there are green marks for users that have submitted their weekly summary:
image
4. log as non-admin user, check if there is no green check marks and the position of dot is in the center.
image
5. submit the weekly summary, log as Owner/Admin/Core Team user to check if the indicator works as expected.

### CODE REVIEW

**Review by EdwinLauM — CHANGES_REQUESTED** [review]
Code looks good and works as intended! As a suggestion, I think it would look nicer to move the red dot of the status to align with the stars and dots.

PR #880

**Review by KurtisIvey — DISMISSED** [review]
Changes function as described. No errors either. Maintains centering on all resolutions. Verified with tester volunteer account. I also agree about the HGN Totals red button, but I know that would have little impact functionality wise. It seems to be perfectly centered when logged in as volunteer. However, with admin login it isn't. 

Image attached verifying functionality as admin:
pr880check

imaged attached verifying functionality as volunteer:
verify

**Review by 20chen-7 — COMMENTED** [review]
The function is intended. Also, I agree with [EdwinLauM](https://github.com/EdwinLauM)'s opinion, the alighment issues.
Login as a volunteer:
Screenshot 2023-06-05 at 11 01 00 AM
Login as adminr:
Screenshot 2023-06-05 at 11 53 42 AM

**Review by eduhorta — COMMENTED** [review]
Hi, @suigunfjc!
The feature works as intended, nice job! 👍

Preview:
Screenshot 2023-06-06 09-11-23

As mentioned by other members, how about align the top red dot of the status with the other dots/stars?

image


Screenshot 2023-06-06 09-13-06

**Review by stevenpaalz — APPROVED** [review]
This feature works as intended. Although see other reviewers note on aligning top red dot with the rest of the column.

Screenshot 2023-06-06 at 2 21 03 PM

**Review by bienzguoa — COMMENTED** [review]
Works as intended! Please see other reviewers note on aligning top red dot with the rest of the column.
Logged as owner:
Screenshot 2023-06-08 at 4 52 14 PM
logged as volunteer:

Screenshot 2023-06-08 at 4 53 13 PM
submitted :
Screenshot 2023-06-08 at 4 58 45 PM
then logged as owner:
Screenshot 2023-06-08 at 4 59 52 PM

**Comment by suigunfjc** [comment]
Hi @EdwinLauM , thank you for pointing this out. My concern here is that the first line is actually never gonna have the check mark as other lines so I didn't change the position of the dot for the first line. I would like to hear opinion from Jae @one-community

**Review by cvtqx — CHANGES_REQUESTED** [review]
Implemented changes work as described.

1. When logged in as volunteer, red dots are centered:
Volunteer - red dots centered

2. After submitting weekly summary on a volunteer account and then logging back in as Admin, I can see a  green checkmark next to the corresponding record. 
Admin-GreenCheckMarkDisplayed

Agreed that the top red dot will look better if aligned with the other dots in the list

**Comment by mahamakhan** [comment]
The features work perfectly fine but the red dot of HGN Totals needs to be aligned.

image
 src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/106139234/b3778ed8-f63f-4b62-8974-bb506a07b34b">

**Review by mahamakhan — DISMISSED** [review]
The features work perfectly fine but the red dot of HGN Totals needs to be aligned.

image
Screenshot 2023-06-09 132933

**Review by OAkinbode — COMMENTED** [review]
Well done. The code works as intended barring the changes others have indicated. 
PR 880 and 383 working

**Review by yufeiStella — DISMISSED** [review]
Good job! It works as intended.
Log in as admin:
Screen Shot 2023-06-09 at 21 32 02
Log in as the volunteer:
Screen Shot 2023-06-09 at 21 30 46

**Comment by one-community** [comment]
> Hi @EdwinLauM , thank you for pointing this out. My concern here is that the first line is actually never gonna have the check mark as other lines so I didn't change the position of the dot for the first line. I would like to hear opinion from Jae @one-community

Good point and catch! Please remove the dot for that line, it doesn't serve a purpose.

**Comment by rachnapandya** [comment]
Except for the improvements others have suggested, the code operates as intended. Good Work.
Log in as a Volunteer.
volunteer_login

Log in as an admin.
admin_login

**Review by Chuehleo — DISMISSED** [review]
Everything works as intended. Good job!
Log in as admin:
Screenshot 2023-06-14 075006
Log in as volunteer:
Screenshot 2023-06-14 074837

**Comment by suigunfjc** [comment]
Hi guys @bienzguoa @20chen-7 @OAkinbode @eduhorta @EdwinLauM @cvtqx @KurtisIvey @mahamakhan @yufeiStella @stevenpaalz @Chuehleo. The latest change was maded to remove the status dot in the first line based on Jae's requirement. 

Also, this PR recover the invisibility state icon in #859, which was affected by the revert commits during the new timer setup (around 2 weeks ago). 

Please test this PR again and remember to pull the latest commit on [BE](https://github.com/OneCommunityGlobal/HGNRest/pull/383) which gets it merged with the latest development branch. Thank you!

Btw, if you have PRs merged during the new timer setup, please check them on the Dev to make sure they weren't mistakely reverted.

File: src/components/LeaderBoard/Leaderboard.jsx [hunk 1]
```diff
@@ -207,23 +208,7 @@ const LeaderBoard = ({
...
-                      width: 15,
-                      height: 15,
-                      borderRadius: 7.5,
-                      margin: 'auto',
-                    }}
-                  />  
-                </Link>
-              </td>
+              <td/>
               <th scope="row">{organizationData.name}</th>
```
Comment (eduhorta) [inline src/components/LeaderBoard/Leaderboard.jsx:212]: Add className="align-middle"

`<th className="align-middle" scope="row">{organizationData.name}</th>`

image

**Review by eduhorta — CHANGES_REQUESTED** [review]
Hi @suigunfjc!
As mentioned by @EdwinLauM and @one-community, the dot for that line wasn't adding up.

The only issue I see is the vertical of HGN Total Members

image

Once you fix this, please tag me or send me a message on Slack so I can approve.

Nice job! 👍

**Comment by suigunfjc** [comment]
Hi @eduhorta, this bug is actually confusing because even with a dev branch in local, the align issue exists: 
image
While this issue doesn't exist in our dev app:
image
So I think the bug is out of this PR's scope and may relate to some css conflicts when you run the app locally.

**Review by eduhorta — APPROVED** [review]
Hey, @suigunfjc!

I think you are right. This behaviour is outside of the scope of this PR.

Everything is working as intended here! Nice job! 👍

**Comment by KaixiangGU** [comment]
Work as intended. GJ!
Screenshot 2023-06-27 at 3 07 24 PM

**Review by KaixiangGU — APPROVED** [review]

**Pull request merged**